src/core/utils/misc.cpp

```diff
@@ -315,30 +317,60 @@ QString Utils::Misc::pythonVersionComplete() {
     return version;
 }
 
-// return best userfriendly storage unit (B, KiB, MiB, GiB, TiB)
+QString Utils::Misc::unitString(Utils::Misc::SizeUnit unit)
+{
+    return QCoreApplication::translate("misc",
+         units[static_cast<int>(unit)].source, units[static_cast<int>(unit)].comment);
+}
+
+// return best userfriendly storage unit (B, KiB, MiB, GiB, TiB, ...)
 // use Binary prefix standards from IEC 60027-2
 // see http://en.wikipedia.org/wiki/Kilobyte
 // value must be given in bytes
 // to send numbers instead of strings with suffixes
-QString Utils::Misc::friendlyUnit(qreal val, bool is_speed)
+bool Utils::Misc::friendlyUnit(qint64 sizeInBytes, qreal &val, Utils::Misc::SizeUnit &unit)
 {
-    if (val < 0)
-        return QCoreApplication::translate("misc", "Unknown", "Unknown (size)");
+    if (sizeInBytes < 0) {
```

Comment:
Try to explain again...
These functions are overloaded because you have decided to give them the same name. But actually, they perform different roles. Old function converts the numeric representation of the size in a string, and it is quite normal to introduce such concept as "wrong" size. It processes this in the usual way, i.e. generates its string representation. Your function, as I said, reduces the number to a certain order. It does not handle the "wrong" size in the usual way, because it is incorrect input for this function, and reports an error (returns false).
In this case, you knowingly allow a function to accept an incorrect input value (using a signed type for input argument), and then "fighting" with this inside it. This is illogical. This behavior is permissible only in cases where passing of incorrect input values into the function cannot be prevented (for example, the input parameter is a string).